examples/navier-stokes/densitycurrent.h

```diff
@@ -123,13 +160,16 @@ static int Exact_DC(CeedInt dim, CeedScalar time, const CeedScalar X[],
   return 0;
 }
 
+// *****************************************************************************
+// TODO: Annotate this function
+// *****************************************************************************
```

Comment:
@LeilaGhaffari @valeriabarra For NS we are mostly pretty good about providing a documentation block for all functions in the QFunction header files, but a few slipped through in the big update. I'm less familiar with these newer functions - would one of you have a bit of time to add documentation blocks to the functions I've marked?

Reply:
Hey @jeremylt ! I have not authored the newer functions, so I would probably not be the best person to help with that. Perhaps @LeilaGhaffari and/or @jedbrown have a clearer idea than me.

Reply:
If we could write the entire rst document interspersed in comments, it might be appropriate, but my inclination is to not try to explain everything in comments because keeping {code, comments, rst} in sync is harder than {code, rst}. So I'd rather just say that someone editing this code should open the sphinx docs and display them side-by-side. Note that the sphinx docs could have `literalinclude` snippets from the source to clarify coding techniques.

Reply:
Maybe we want just a single description sentence for these then with the docs providing full details? The lack of a description meant I had to play a bit of hide and seek to figure out what was going on as I was trying to get these files working on CUDA.

Reply:
I agree with @jedbrown . It's hard to sync code, comments and rst while the comments are not really as helpful as rst.

Reply:
I second @jeremylt on the fact that at least a concise sentence before these functions saying at least why now we have an "exact" version wouldn't hurt.

Reply:
Yeah, I try to add a one line description to my functions to help future me and new collaborators. I've refractored stuff too many times to feel comfortable without something like that